File: csharp/Behavioral/14-Command/Command/AssignHandymanCommand.cs
public class AssignHandymanCommand : ICommand
{
    private readonly JobCard _jobCard;
    private readonly string  _handyman;
    private string?          _previousAssignee;
    private string           _previousStatus = string.Empty;

    public AssignHandymanCommand(JobCard jobCard, string handyman)
    {
        _jobCard  = jobCard;
        _handyman = handyman;
    }

    public void Execute()
    {
        _previousAssignee = _jobCard.Assignee;
        _previousStatus   = _jobCard.Status;
        _jobCard.Assignee = _handyman;
        _jobCard.Status   = "Assigned";
        Console.WriteLine($"  Assigned '{_handyman}' to {_jobCard.JobId}. Status → Assigned");
    }

    public void Undo()
    {
        _jobCard.Assignee = _previousAssignee;
        _jobCard.Status   = _previousStatus;
        Console.WriteLine($"  Undo assign: {_jobCard.JobId} Assignee → {_previousAssignee ?? "none"}, Status → {_previousStatus}");
    }
}

File: typescript/Behavioral/14-Command/Command/AssignHandymanCommand.ts
import { ICommand } from "./ICommand";
import { JobCard } from "../Model/JobCard";

export class AssignHandymanCommand implements ICommand {
  private previousAssignee: string | null = null;
  private previousStatus: string = "";

  constructor(
    private readonly jobCard: JobCard,
    private readonly handyman: string,
  ) {}

  execute(): void {
    this.previousAssignee = this.jobCard.assignee;
    this.previousStatus = this.jobCard.status;
    this.jobCard.assignee = this.handyman;
    this.jobCard.status = "Assigned";
    console.log(
      `  Assigned '${this.handyman}' to ${this.jobCard.jobId}. Status → Assigned`,
    );
  }

  undo(): void {
    this.jobCard.assignee = this.previousAssignee;
    this.jobCard.status = this.previousStatus;
    console.log(
      `  Undo assign: ${this.jobCard.jobId} Assignee → ${this.previousAssignee ?? "none"}, Status → ${this.previousStatus}`,
    );
  }
}
